File: java codes/QuickSort.java
import java.util.Scanner;

public class QuickSort {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");

        int length = sc.nextInt();
        int[] intArray = new int[length];

        System.out.print("Enter the array elements: ");

        for (int i = 0; i < intArray.length; i++) {
            intArray[i] = sc.nextInt();
        }

        quickSort(intArray, 0, intArray.length);

        System.out.print("Sorted array: ");
        for (int i = 0; i < intArray.length; i++) {
            System.out.print(intArray[i] + " ");
        }
    }

    public static void quickSort(int[] input, int start, int end) {

        if(end - start < 2) {
            return;
        }

        int pivotIndex = partition(input, start, end);
        quickSort(input, start, pivotIndex);
        quickSort(input, pivotIndex + 1, end);
    }

    public static int partition(int[] input, int start, int end) {
        
        // This is using the first element as the pivot
        int pivot = input[start];
        int i = start;
        int j = end;

        while (i < j) {

            while (i < j && input[--j] >= pivot);
            if(i < j) {
                input[i] = input[j];
            }

            while (i < j && input[++i] <= pivot);
            if(i < j) {
                input[j] = input[i];
            }
        }

        input[j] = pivot;
        return j;

    }
}

File: python-sorting-algorithms/Quick_Sort.py
a = [5, 4, 8, 3, 4, 14, 90, 45, 9, 3, 2, 4, 45]
p, r = 0, len(a)-1
def quick_sort(a, p, r):
    if p < r:
        q = partition(a, p, r)
        quick_sort(a, p, q-1)
        quick_sort(a, q+1, r)
def partition(a, p, r):
    x = a[r]
    i = p-1
    for j in range(p,r):
        if a[j] <= x:
            i += 1
            print(a[i],a[j])
            a[i], a[j] = a[j], a[i]
    a[i+1], a[r] = a[r], a[i+1]
    return i+1
quick_sort(a, p, r)
print(a)